ocaml/xenopsd/xc/xenops_server_xen.ml

```diff
@@ -2036,15 +2036,15 @@ module VM = struct
                   Device.Dm.Disabled
             in
             let parallel = List.assoc_opt "parallel" vm.Vm.platformdata in
+            let tpm = Option.fold ~none:hvm_info.tpm ~some:Option.some vtpm in
```

Comment:
Is this the same as `Option.value ~default:hvm_info.tpm vtpm`? I'd find that easier to read than the fold.

Reply:
Also would be useful to add a log line here that shows the final outcome of the 'tpm' object (e.g. its uuid it it exists)

Reply:
> Is this the same as Option.value ~default:hvm_info.tpm vtpm?
Not exactly, we don't want to unwrap the value, which is what `Option.value` does.
This case can be thought of as an if is none, give a default option value, otherwise leave it unchanged